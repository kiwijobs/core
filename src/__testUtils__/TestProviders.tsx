import React, { FC, Suspense } from 'react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../theme';

export const TestProviders: FC<{}> = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <Suspense fallback="mocked loading...">{children}</Suspense>
    </ThemeProvider>
  );
};

export const wrapper = TestProviders; // React Testing Library friendly name.
