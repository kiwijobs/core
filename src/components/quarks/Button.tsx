import React, { forwardRef } from 'react';
import { Box, BoxProps } from './Reflexbox';

export const Button = forwardRef<HTMLButtonElement, BoxProps>(({ sx, ...props }, ref) => (
  <Box
    ref={ref}
    as={'button' as 'button'}
    type="button"
    sx={{
      cursor: 'pointer',
      color: 'inherit',
      bg: 'transparent',
      transition: 'all 150ms ease-in-out',
      textAlign: 'center',
      position: 'relative',
      ' -webkit-tap-highlight-color': 'transparent',
      boxShadow: 'none',
      outline: 'none',

      '&:disabled': {
        pointerEvents: 'none',
      },

      ...sx,
    }}
    {...props}
  />
));
