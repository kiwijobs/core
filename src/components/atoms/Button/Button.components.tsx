import { SXProp } from '../../quarks';

export const Primary: Partial<SXProp> = {
  fontScale: 3,
  fontWeight: 'bold',
  textTransform: 'uppercase',
  py: 2,
  px: 6,
  minWidth: '14rem',
  bg: 'primary',
  color: 'white',
  borderRadius: '2rem',

  '&:hover': { bg: 'primary.90' },
  '&:disabled': { bg: 4, color: 2 },
};

export const Secondary: Partial<SXProp> = {
  ...Primary,
  bg: 'white',
  color: 2,

  '&:hover': { color: 1 },
  '&:disabled': { color: 3 },
};
