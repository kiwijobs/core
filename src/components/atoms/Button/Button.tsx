import React, { forwardRef } from 'react';
import { Box, BoxProps } from '../../quarks';
import { Loader } from '../Loader';

export interface ButtonProps extends BoxProps<HTMLButtonElement> {
  loading?: boolean;
  variant: Pick<BoxProps, 'theme'>;
}

export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
  ({ children, loading = false, ...props }, ref) => (
    <Box ref={ref} tx="buttons" as={'button' as 'button'} type="button" {...props}>
      {loading ? <Loader /> : children}
    </Box>
  )
);
