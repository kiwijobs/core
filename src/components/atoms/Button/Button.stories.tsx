import React from 'react';
import { storiesOf } from '@storybook/react';
import { Button } from './Button';
import { Space } from '../Grid';

storiesOf('Atoms|Button', module)
  .add('Standard', () => (
    <Space my={3} mx={2}>
      <Button variant="Primary">Zapisz</Button>
      <Button variant="Primary" disabled>
        Disabled
      </Button>
      <Button variant="Primary" loading>
        Zapisz
      </Button>
      <Button variant="Primary" disabled loading>
        Disabled
      </Button>
      <br />
      <Button variant="Secondary">Anuluj</Button>
      <Button variant="Secondary" disabled>
        Disabled
      </Button>
      <Button variant="Secondary" loading>
        Anuluj
      </Button>
      <Button variant="Secondary" loading disabled>
        Disabled
      </Button>
      <br />
      <Button
        sx={{
          fontScale: 1,
          bg: 'secondary',
          '&:hover': { bg: 'secondary.90' },
        }}
        variant="Primary"
      >
        Primary
      </Button>
      <Button
        sx={{
          fontScale: 1,
          bg: 'tangerine',
          '&:hover': { bg: 'tangerine.90' },
        }}
        variant="Primary"
      >
        Primary
      </Button>
      <Button
        sx={{
          fontScale: 1,
          bg: 'pink',
          '&:hover': { bg: 'pink.90' },
        }}
        variant="Primary"
      >
        Primary
      </Button>
      <Button
        sx={{
          fontScale: 1,
          bg: 'dark',
          '&:hover': { bg: 'dark.120' },
        }}
        variant="Primary"
      >
        Primary
      </Button>
      <br />
      <Button
        sx={{
          fontScale: 1,
          bg: 'secondary',
          '&:hover': { bg: 'secondary.90' },
        }}
        variant="Primary"
        loading
      >
        Primary
      </Button>
      <Button
        sx={{
          fontScale: 1,
          bg: 'tangerine',
          '&:hover': { bg: 'tangerine.90' },
        }}
        variant="Primary"
        loading
      >
        Primary
      </Button>
      <Button
        sx={{
          fontScale: 1,
          bg: 'pink',
          '&:hover': { bg: 'pink.90' },
        }}
        variant="Primary"
        loading
      >
        Primary
      </Button>
      <Button
        sx={{
          fontScale: 1,
          bg: 'dark',
          '&:hover': { bg: 'dark.120' },
        }}
        variant="Primary"
        loading
      >
        Primary
      </Button>
    </Space>
  ))
  .add('Baseline', () => (
    <Space m={4}>
      <Button>HTML Button Element</Button>
    </Space>
  ))
  .add('Custom', () => (
    <Space my={3} mx={2}>
      <Button
        sx={{
          border: 1,
          borderRadius: 1,
          bg: 'white',
          color: 2,
          px: 3,
          py: 2,
          '&:hover': { bg: 4 },
        }}
      >
        Custom HTML Button
      </Button>
    </Space>
  ));
