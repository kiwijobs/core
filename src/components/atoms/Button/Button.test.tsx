import React from 'react';
import { wrapper } from '../../../__testUtils__';
import { Button } from './Button';
import { render } from '@testing-library/react';

describe('Button', () => {
  it('Supports variants', () => {
    expect(render(<Button variant="Primary" />, { wrapper }).container).toMatchSnapshot();
    expect(render(<Button variant="Secondary" />, { wrapper }).container).toMatchSnapshot();
    expect(render(<Button variant="Text" />, { wrapper }).container).toMatchSnapshot();
    expect(render(<Button variant="Link" />, { wrapper }).container).toMatchSnapshot();
  });

  it('Supports "loading" prop', () => {
    expect(render(<Button loading>Stuff</Button>, { wrapper }).container).toMatchSnapshot();
  });
});
