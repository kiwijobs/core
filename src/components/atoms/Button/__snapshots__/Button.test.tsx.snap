// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Supports "loading" prop 1`] = `
.c0 {
  cursor: pointer;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  z-index: 2;
  top: calc(50% - 0.714em / 2);
  left: calc(50% - 0.714em / 2);
}

.c2 {
  border-radius: 50%;
  border: calc(0.714em / 2) solid;
  width: 0.714em;
  height: 0.714em;
  display: inline-block;
  margin-left: calc(0.714em / 4);
  margin-right: calc(0.714em / 4);
}

.c3 {
  -webkit-animation: ghjfcd 1.4s infinite ease-in-out both;
  animation: ghjfcd 1.4s infinite ease-in-out both;
}

.c3:nth-of-type(1) {
  -webkit-animation-delay: -0.32s;
  animation-delay: -0.32s;
}

.c3:nth-of-type(2) {
  -webkit-animation-delay: -0.16s;
  animation-delay: -0.16s;
  opacity: 0.5;
}

<div>
  <button
    class="c0"
    type="button"
  >
    <div
      class="c1"
    >
      <div
        class="c2 c3"
      />
      <div
        class="c2 c3"
      />
      <div
        class="c2 c3"
      />
    </div>
  </button>
</div>
`;

exports[`Button Supports variants 1`] = `
.c0 {
  cursor: pointer;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 40px;
  padding-right: 40px;
  min-width: 14rem;
  background-color: rgb(8,191,130);
  color: rgb(255,255,255);
  border-radius: 2rem;
  font: 1.4rem / 2.4rem "Open Sans";
  font-weight: bold;
}

.c0:hover {
  background-color: rgb(7,171,117);
}

.c0:disabled {
  background-color: rgb(228,230,237);
  color: rgb(133,136,149);
}

<div>
  <button
    class="c0"
    type="button"
  />
</div>
`;

exports[`Button Supports variants 2`] = `
.c0 {
  cursor: pointer;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 40px;
  padding-right: 40px;
  min-width: 14rem;
  background-color: rgb(255,255,255);
  color: rgb(133,136,149);
  border-radius: 2rem;
  font: 1.4rem / 2.4rem "Open Sans";
  font-weight: bold;
}

.c0:hover {
  color: rgb(39,50,55);
}

.c0:disabled {
  color: rgb(204,206,220);
}

<div>
  <button
    class="c0"
    type="button"
  />
</div>
`;

exports[`Button Supports variants 3`] = `
.c0 {
  cursor: pointer;
  display: inline-block;
  color: rgb(133,136,149);
  font: 1.2rem / 1.6rem "Open Sans";
}

.c0:hover {
  color: rgb(39,50,55);
}

.c0:disabled {
  color: rgb(204,206,220);
}

<div>
  <button
    class="c0"
    type="button"
  />
</div>
`;

exports[`Button Supports variants 4`] = `
.c0 {
  cursor: pointer;
  display: inline;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0:disabled {
  color: rgb(204,206,220);
  pointer-events: none;
}

<div>
  <button
    class="c0"
    type="button"
  />
</div>
`;
